src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -1654,7 +1654,7 @@ trait Namers extends MethodSynthesis {
         if (sym hasAnnotation NativeAttr)
           sym resetFlag DEFERRED
         else {
-          if (!symbolAllowsDeferred && ownerRequiresConcrete) fail(AbstractVar)
+          if (!symbolAllowsDeferred && ownerRequiresConcrete && !sym.isJava) fail(AbstractVar)
```

Comment:
[getting back to this now, please forgive the delay]
What I don't understand is why adding doc comments changes things. Normally, Java code goes through namer and typer when mixed Scala-Java projects are used. Therefore, retaining the doc comments should not change anything at all.